corehq/apps/reports/generic.py

```diff
@@ -660,9 +660,15 @@ def export_response(self):
             export_all_rows_task.delay(self.__class__, self.__getstate__())
             return HttpResponse()
         else:
-            temp = io.BytesIO()
-            export_from_tables(self.export_table, temp, self.export_format)
-            return export_response(temp, self.export_format, self.export_name)
+            # We only want to cache the responses which serve files directly
+            # The response which return 200 and emails the reports should not be cached
+            return self._export_response_direct()
```

Comment:
Looks like you forgot to remove `@request_cache()` decorator on line 652 above

https://github.com/dimagi/commcare-hq/blob/442632da65214e959235bab6f776e4e2be1a7b0b/corehq/apps/reports/generic.py#L652

Reply:
Ohh! Yes. I think it was added while I merged with a master in the last commit. Thanks, @millerdev for catching that 😄

Reply:
Sorry for dragging this out, but considering that the tests did not catch this error, would it be worth adding a test that verifies correct caching behavior since that's what's being changed here? Specifically, it should verify that the response is not cached if `self.exportable_all` is true. Probably should also test that the response _is_ cached in this `else` block.